### PR TITLE
feat(wallet) add recipients API and integrate with filter

### PR DESCRIPTION
**Pull request by stefandunca**
### Updates #10025

Bumps [status-go](https://github.com/status-im/status-go/pull/3623) to include the new API.

Add a new RecipientsModel to the activity Controller.
Extends the activity Controller with API to access and manage data in the RecipientsModel.

Resolving addresses to names remains to be implemented.

### CODE REVIEW

File: src/app/modules/main/wallet_section/activity/controller.nim [hunk 1]
```diff
@@ -291,4 +301,22 @@ QtObject:
...
+  proc updateRecipientsModel*(self: Controller) {.slot.} =
+    let response = backend_activity.getAllRecipients(0, FETCH_RECIPIENTS_BATCH_COUNT_DEFAULT)
+    if response.error != nil:
+      error "error fetching recipients: ", response.error
+      return
+
+    let result = json.to(response.result, backend_activity.GetAllRecipientsResponse)
+    self.recipientsModel.addAddresses(result.addresses, 0, result.hasMore)
+
+  proc loadMoreRecipients(self: Controller) {.slot.} =
```
Comment (Khushboo-dev-cpp) [inline src/app/modules/main/wallet_section/activity/controller.nim:315]: Hey @stefandunca why is this needed?
will the user only get list of recipient based on transactions loaded?
Also will they be exlusive and non repeating?

Comment (stefandunca) [reply src/app/modules/main/wallet_section/activity/controller.nim:315]: Because we can't control how many transactions might be in the DB, there is no limit to how many recipients we can have, so this way, we set a limit to the retrieve (2000 for now) and have the option to retrieve the rest later on. Having something more than lazy loading of batches was too much. However, if having more than a few hundred is expected and not an extreme corner case, we can extend this with search and lazy loading like the activity history.

For now, don't use it. Load the first batch, and let's see if that is enough (based on testing with famous accounts). The next improvement will be retrieving asynchronous, leaving it for the next cycle.

**Review by Khushboo-dev-cpp — APPROVED** [review]
LGTM! thank for the quick fix
